packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1588,11 +1599,23 @@ export class ContainerRuntime
 			snapshotWithContents,
 		} = context;
 
+		this.clientDetails = clientDetails;
+		this.isSummarizerClient = this.clientDetails.type === summarizerClientType;
+
+		// In old loaders without dispose functionality, closeFn is equivalent but will also switch container to readonly mode
+		this.disposeFn = disposeFn ?? closeFn;
```

Comment:
It'll be so great to NEVER have ambiguous code like this again, where it's a mystery whether we can remove the fallback or not.